step/skip_testing.go

```diff
@@ -0,0 +1,193 @@
+// tha standard JSON marshalling would escape "\" as "\\".
+type TestPathString string
+
+func (cs TestPathString) MarshalJSON() ([]byte, error) {
+	escaped := `"` + string(cs) + `"`
+	return []byte(escaped), nil
+}
+
+func (b XcodebuildBuilder) findTestPlan(testPlan, projectPath string) (string, error) {
+	return b.fileManager.FindFile(filepath.Dir(projectPath), testPlan+".xctestplan")
+}
+
+func (b XcodebuildBuilder) backupTestPlan(testPlanPath string) (string, error) {
+	tmpDir, err := b.pathProvider.CreateTempDir("backupTestPlans")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temp dir for backup: %w", err)
+	}
+
+	backupTestPlanPath := filepath.Join(tmpDir, filepath.Base(testPlanPath))
+
+	testPlanContent, err := b.fileManager.ReadFile(testPlanPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read original test plan: %w", err)
+	}
+
+	if err := b.fileManager.WriteFile(backupTestPlanPath, testPlanContent, 0644); err != nil {
+		return "", fmt.Errorf("failed to write backup test plan: %w", err)
+	}
+
+	return backupTestPlanPath, nil
+}
+
+func (b XcodebuildBuilder) addSkippedTestsToTestPlanFile(testPlanPath string, skipTesting []string) error {
+	skipTestingInTargets, err := b.parseSkipTestingFormat(skipTesting)
+	if err != nil {
+		return fmt.Errorf("failed to parse skip testing format: %w", err)
+	}
+
+	testPlan, err := b.readAndParseTestPlan(testPlanPath)
+	if err != nil {
+		return fmt.Errorf("failed to read and parse test plan: %w", err)
+	}
+
+	updatedTestPlan, err := b.addSkippedTestsToTestPlan(testPlan, skipTestingInTargets)
+	if err != nil {
+		return fmt.Errorf("failed to add skipped tests to test plan: %w", err)
+	}
+
+	if err := b.writeTestPlan(testPlanPath, updatedTestPlan); err != nil {
+		return fmt.Errorf("failed to write updated test plan: %w", err)
+	}
+
+	return nil
+}
+
+func (b XcodebuildBuilder) restoreTestPlan(backupTestPlanPath, originalTestPlanPath string) error {
+	backupTestPlanContent, err := b.fileManager.ReadFile(backupTestPlanPath)
+	if err != nil {
+		return fmt.Errorf("failed to read backup test plan: %w", err)
+	}
+
+	if err := b.fileManager.WriteFile(originalTestPlanPath, backupTestPlanContent, 0644); err != nil {
+		return fmt.Errorf("failed to restore original test plan: %w", err)
+	}
+
+	b.logger.Printf("Original test plan restored from backup: %s", originalTestPlanPath)
+	return nil
+}
+
+func (b XcodebuildBuilder) writeTestPlan(testPlanPath string, updatedTestPlan map[string]interface{}) error {
+	updatedTestPlanContent, err := json.MarshalIndent(updatedTestPlan, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal updated test plan to JSON: %w", err)
+	}
+
+	if err := b.fileManager.WriteFile(testPlanPath, updatedTestPlanContent, 0644); err != nil {
+		return fmt.Errorf("failed to write updated test plan to file: %w", err)
+	}
+
+	return nil
+}
+
+func (b XcodebuildBuilder) parseSkipTestingFormat(skipTesting []string) (map[string][]TestPathString, error) {
```

Comment:
A test case would be nice to have.

Reply:
Added here: https://github.com/bitrise-steplib/steps-xcode-build-for-test/pull/74/commits/9efb86374919a397e9309ff63e0f0dc97622e1ff